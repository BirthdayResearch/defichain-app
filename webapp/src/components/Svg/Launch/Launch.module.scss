// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../../scss/variables';
@import '../../../scss/bootstrap/variables';

.launch {
  color: $primary !important;
  --duration: 2s;
  --logoease: ease-out;
  --groundease: ease-out;
  --branchease: ease-out;
}

.notAnimated {
  --duration: 0s;
}

.whole_logo_group {
  transform-origin: 54px 54px;
  animation: logo_rotate_anim var(--duration) 1 var(--logoease) forwards;
}

.right_crescent_mask_path {
  animation: right_crescent_mask_path_anim var(--duration) 1 var(--groundease)
    forwards;
}

.right_rear_hemi_mask_path {
  animation: right_rear_hemi_mask_path_anim var(--duration) 1 var(--groundease)
    forwards;
}

.branch001_left,
.branch001_right {
  animation: branch001_anim var(--duration) 1 var(--branchease) forwards;
}

.branch002_left_A,
.branch002_left_B,
.branch002_right_A,
.branch002_right_B {
  animation: branch002_anim var(--duration) 1 var(--branchease) forwards;
}

.branch003_left_A,
.branch003_left_D,
.branch003_right_A,
.branch003_right_D {
  animation: branch003_AD_anim var(--duration) 1 var(--branchease) forwards;
}

.branch003_left_B,
.branch003_left_C,
.branch003_right_B,
.branch003_right_C {
  animation: branch003_BC_anim var(--duration) 1 var(--branchease) forwards;
}

@keyframes logo_rotate_anim {
  0% {
    transform: rotate(120deg);
  }

  23% {
    transform: rotate(90deg);
  }
  25% {
    transform: rotate(90deg);
  }

  49% {
    transform: rotate(60deg);
  }
  51% {
    transform: rotate(60deg);
  }

  75% {
    transform: rotate(30deg);
  }
  77% {
    transform: rotate(30deg);
  }

  100% {
    transform: rotate(0deg);
  }
}

@keyframes right_crescent_mask_path_anim {
  0% {
    transform: translateX(54px);
  }
  23% {
    transform: translateX(0px);
  }
  100% {
    transform: translateX(0px);
  }
}

@keyframes right_rear_hemi_mask_path_anim {
  0% {
    transform: translateX(54px);
  }

  23% {
    transform: translateX(0px);
  }
  25% {
    transform: translateX(-10px);
  }

  49% {
    transform: translateX(-20px);
  }
  51% {
    transform: translateX(-20px);
  }

  75% {
    transform: translateX(-30px);
  }
  77% {
    transform: translateX(-30px);
  }

  100% {
    transform: translateX(-40px);
  }
}

@keyframes branch001_anim {
  0% {
    stroke-dashoffset: 40;
  }
  25% {
    stroke-dashoffset: 40;
  }
  51% {
    stroke-dashoffset: 0;
  }
  100% {
    stroke-dashoffset: 0;
  }
}

@keyframes branch002_anim {
  0% {
    stroke-dashoffset: 40;
  }
  51% {
    stroke-dashoffset: 40;
  }
  77% {
    stroke-dashoffset: 6;
  }
  100% {
    stroke-dashoffset: 6;
  }
}

@keyframes branch003_AD_anim {
  0% {
    stroke-dashoffset: 35;
  }
  78% {
    stroke-dashoffset: 35;
  }
  100% {
    stroke-dashoffset: 10;
  }
}

@keyframes branch003_BC_anim {
  0% {
    stroke-dashoffset: 30;
  }
  78% {
    stroke-dashoffset: 30;
  }
  100% {
    stroke-dashoffset: 2;
  }
}
