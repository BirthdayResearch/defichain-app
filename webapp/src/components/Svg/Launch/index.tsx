import React from 'react';
import style from './Launch.module.scss';

interface LaunchLogoProps {
  isNotAnimated?: boolean;
}

const Launch = (props: LaunchLogoProps) => {
  const { isNotAnimated } = props;
  return (
    <svg
      width='108'
      height='108'
      xmlns='http://www.w3.org/2000/svg'
      xmlnsXlink='http://www.w3.org/1999/xlink'
      viewBox='0 0 108 108'
      version='1.1'
      className={`${style.launch} ${!isNotAnimated ? '' : style.notAnimated}`}
    >
      <defs>
        <clipPath id='right_crescent_mask'>
          <path
            className={style.right_crescent_mask_path}
            opacity='1'
            fill='#f0f'
            stroke='none'
            d='M54-9.553h54v125.768H54z'
          />
        </clipPath>
        <clipPath id='right_rear_hemi_mask'>
          <path
            className={style.right_rear_hemi_mask_path}
            opacity='1'
            fill='#f0f'
            stroke='none'
            d='M54-9.553h54v125.768H54z'
          />
        </clipPath>
        <clipPath id='outside_clip_mask'>
          <circle
            id='outside_clip_circle'
            cy='53.795'
            cx='54.043'
            r='53.92'
            opacity='1'
            fill='#f0f'
            stroke='none'
          />
        </clipPath>
      </defs>
      <g className={style.whole_logo_group} clipPath='url(#outside_clip_mask)'>
        <path
          id='right_rear_hemi'
          clipPath='url(#right_rear_hemi_mask)'
          d='M54.044-.125v107.84a53.92 53.92 0 0053.919-53.919A53.92 53.92 0 0054.044-.124z'
          fill='currentColor'
          stroke='none'
        />
        <path
          id='front_crescent'
          clipPath='url(#right_crescent_mask)'
          d='M54.042-.125v11.927a41.994 41.994 0 01.002 0 41.994 41.994 0 0141.994 41.994 41.994 41.994 0 01-41.994 41.992 41.994 41.994 0 01-.002 0v11.927a53.92 53.92 0 00.002 0 53.92 53.92 0 0053.919-53.919A53.92 53.92 0 0054.044-.124a53.92 53.92 0 00-.002 0z'
          fill='currentColor'
          stroke='none'
        />
        <g
          id='branches_group'
          fill='none'
          stroke='currentColor'
          strokeWidth='11.982'
          strokeLinecap='butt'
          strokeLinejoin='miter'
          strokeMiterlimit='4'
          strokeDasharray='none'
          strokeOpacity='1'
          markerStart='none'
          markerMid='none'
          markerEnd='none'
        >
          <path
            className={style.branch001_left}
            d='M58.635 49.087L39.621 68.542'
            strokeDasharray='40'
            strokeDashoffset='40'
          />
          <path
            className={style.branch001_right}
            d='M58.635 58.516L39.621 39.061'
            strokeDasharray='40'
            strokeDashoffset='40'
          />
          <path
            className={style.branch002_left_A}
            d='M41.296 64.613L40.38 90.12'
            strokeDasharray='40'
            strokeDashoffset='40'
          />
          <path
            className={style.branch002_left_B}
            d='M43.785 66.696l-25.858.555'
            strokeDasharray='40'
            strokeDashoffset='40'
          />
          <path
            className={style.branch002_right_A}
            d='M43.785 40.907l-25.858-.555'
            strokeDasharray='40'
            strokeDashoffset='40'
          />
          <path
            className={style.branch002_right_B}
            d='M41.296 42.99l-.916-25.509'
            strokeDasharray='40'
            strokeDashoffset='40'
          />
          <path
            className={style.branch003_left_A}
            d='M40.159 87.883l7.947 23.092'
            strokeDasharray='35'
            strokeDashoffset='35'
          />
          <path
            className={style.branch003_left_B}
            d='M43.575 85.028L18.906 97.85'
            strokeDasharray='30'
            strokeDashoffset='30'
          />
          <path
            className={style.branch003_left_C}
            d='M23.136 63.977L9.684 89.27'
            strokeDasharray='30'
            strokeDashoffset='30'
          />
          <path
            className={style.branch003_left_D}
            d='M20.047 67.555L-3.96 59.53'
            strokeDasharray='35'
            strokeDashoffset='35'
          />
          <path
            className={style.branch003_right_A}
            d='M20.047 40.048L-3.96 48.072'
            strokeDasharray='35'
            strokeDashoffset='35'
          />
          <path
            className={style.branch003_right_B}
            d='M23.136 43.625L9.684 18.334'
            strokeDasharray='30'
            strokeDashoffset='30'
          />
          <path
            className={style.branch003_right_C}
            d='M43.575 22.575L18.906 9.752'
            strokeDasharray='30'
            strokeDashoffset='30'
          />
          <path
            className={style.branch003_right_D}
            d='M40.159 19.72l7.947-23.092'
            strokeDasharray='35'
            strokeDashoffset='35'
          />
        </g>
      </g>
    </svg>
  );
};

export default Launch;
