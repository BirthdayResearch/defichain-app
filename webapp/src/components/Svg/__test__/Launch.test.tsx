import React from 'react';
import Launch from '../Launch';
import { shallow } from 'enzyme';

describe('Launch component', () => {
  it('should check for snapshot', () => {
    const wrapper = shallow(<Launch />);
    expect(wrapper).toMatchSnapshot();
  });
});
