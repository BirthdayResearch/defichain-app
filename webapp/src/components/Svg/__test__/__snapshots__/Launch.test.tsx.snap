// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch component should check for snapshot 1`] = `
<svg
  className="undefined "
  height="108"
  version="1.1"
  viewBox="0 0 108 108"
  width="108"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <clipPath
      id="right_crescent_mask"
    >
      <path
        d="M54-9.553h54v125.768H54z"
        fill="#f0f"
        opacity="1"
        stroke="none"
      />
    </clipPath>
    <clipPath
      id="right_rear_hemi_mask"
    >
      <path
        d="M54-9.553h54v125.768H54z"
        fill="#f0f"
        opacity="1"
        stroke="none"
      />
    </clipPath>
    <clipPath
      id="outside_clip_mask"
    >
      <circle
        cx="54.043"
        cy="53.795"
        fill="#f0f"
        id="outside_clip_circle"
        opacity="1"
        r="53.92"
        stroke="none"
      />
    </clipPath>
  </defs>
  <g
    clipPath="url(#outside_clip_mask)"
  >
    <path
      clipPath="url(#right_rear_hemi_mask)"
      d="M54.044-.125v107.84a53.92 53.92 0 0053.919-53.919A53.92 53.92 0 0054.044-.124z"
      fill="currentColor"
      id="right_rear_hemi"
      stroke="none"
    />
    <path
      clipPath="url(#right_crescent_mask)"
      d="M54.042-.125v11.927a41.994 41.994 0 01.002 0 41.994 41.994 0 0141.994 41.994 41.994 41.994 0 01-41.994 41.992 41.994 41.994 0 01-.002 0v11.927a53.92 53.92 0 00.002 0 53.92 53.92 0 0053.919-53.919A53.92 53.92 0 0054.044-.124a53.92 53.92 0 00-.002 0z"
      fill="currentColor"
      id="front_crescent"
      stroke="none"
    />
    <g
      fill="none"
      id="branches_group"
      markerEnd="none"
      markerMid="none"
      markerStart="none"
      stroke="currentColor"
      strokeDasharray="none"
      strokeLinecap="butt"
      strokeLinejoin="miter"
      strokeMiterlimit="4"
      strokeOpacity="1"
      strokeWidth="11.982"
    >
      <path
        d="M58.635 49.087L39.621 68.542"
        strokeDasharray="40"
        strokeDashoffset="40"
      />
      <path
        d="M58.635 58.516L39.621 39.061"
        strokeDasharray="40"
        strokeDashoffset="40"
      />
      <path
        d="M41.296 64.613L40.38 90.12"
        strokeDasharray="40"
        strokeDashoffset="40"
      />
      <path
        d="M43.785 66.696l-25.858.555"
        strokeDasharray="40"
        strokeDashoffset="40"
      />
      <path
        d="M43.785 40.907l-25.858-.555"
        strokeDasharray="40"
        strokeDashoffset="40"
      />
      <path
        d="M41.296 42.99l-.916-25.509"
        strokeDasharray="40"
        strokeDashoffset="40"
      />
      <path
        d="M40.159 87.883l7.947 23.092"
        strokeDasharray="35"
        strokeDashoffset="35"
      />
      <path
        d="M43.575 85.028L18.906 97.85"
        strokeDasharray="30"
        strokeDashoffset="30"
      />
      <path
        d="M23.136 63.977L9.684 89.27"
        strokeDasharray="30"
        strokeDashoffset="30"
      />
      <path
        d="M20.047 67.555L-3.96 59.53"
        strokeDasharray="35"
        strokeDashoffset="35"
      />
      <path
        d="M20.047 40.048L-3.96 48.072"
        strokeDasharray="35"
        strokeDashoffset="35"
      />
      <path
        d="M23.136 43.625L9.684 18.334"
        strokeDasharray="30"
        strokeDashoffset="30"
      />
      <path
        d="M43.575 22.575L18.906 9.752"
        strokeDasharray="30"
        strokeDashoffset="30"
      />
      <path
        d="M40.159 19.72l7.947-23.092"
        strokeDasharray="35"
        strokeDashoffset="35"
      />
    </g>
  </g>
</svg>
`;
